types/tx_msg.go

```diff
@@ -2,45 +2,59 @@ package types
 
 import (
 	"encoding/json"
+
+	"github.com/tendermint/tendermint/crypto"
 )
 
-// Transactions messages must fulfill the Msg
-type Msg interface {
+type (
+	// Msg defines the interface a transaction message must fulfill.
+	Msg interface {
 
-	// Return the message type.
-	// Must be alphanumeric or empty.
-	Route() string
+		// Return the message type.
+		// Must be alphanumeric or empty.
+		Route() string
 
-	// Returns a human-readable string for the message, intended for utilization
-	// within tags
-	Type() string
+		// Returns a human-readable string for the message, intended for utilization
+		// within tags
+		Type() string
 
-	// ValidateBasic does a simple validation check that
-	// doesn't require access to any other information.
-	ValidateBasic() error
+		// ValidateBasic does a simple validation check that
+		// doesn't require access to any other information.
+		ValidateBasic() error
 
-	// Get the canonical byte representation of the Msg.
-	GetSignBytes() []byte
+		// Get the canonical byte representation of the Msg.
+		GetSignBytes() []byte
```

Comment:
Let's leave a TODO here to deprecate this?

Reply:
We'd need to deprecate/remove a lot IMHO (e.g. TxBuilder, lots of utils, interfaces, etc..). I'd rather have a followup issue that lays all of that out.

Reply:
A follow-up issue is fine.